api-remove.go

```diff
@@ -195,6 +195,13 @@ func (c Client) RemoveObjectsWithContext(ctx context.Context, bucketName string,
 				contentMD5Base64: sumMD5Base64(removeBytes),
 				contentSHA256Hex: sum256Hex(removeBytes),
 			})
+			if resp != nil {
```

Comment:
Delete Multiple Objects REST API returns 200 OK even when request parameters (headers and query params included) are invalid. See `Example 3: Malformed XML in the Request` in https://docs.aws.amazon.com/AmazonS3/latest/API/multiobjectdeleteapi.html.

We have to fix minio server too to send 200 OK with error message as XML body, but that will be a different PR. So, in this change we need to handle server's response as documented above.
(N B this is something I found (only) when I was reviewing this change.)

Reply:
@krisis , server already sends error message as xml body along with 200 OK. The httpRespToErrorResponse method decodes the error response from the xml body.

Reply:
It is wrong to blindly decoding response body without checking HTTP status code.  I believe multiple delete response XML is sent only on status code = 200.

Reply:
@krisis, my bad - server actually sends a 404 The specified key does not exist.